### PR TITLE
attribute definitions create and delete

### LINKED ISSUE #132: Finish attribute definition management
* The goal of this task is to finish the page for managing attribute definitions in the admin section.
* There should not be displayed core attributes.
* It should be possible to create and delete already existing attributes.

### PR DESCRIPTION
**Pull request by DanoFecko**
resolves #132 

* users are now able to CREATE Attribute Definition
* users are now able to DELETE Attribute Definition

### CODE REVIEW

**Comment by DanoFecko** [comment]
@xkureck @xkostka2 take a look, please

File: src/assets/i18n/en.json [hunk 1]
```diff
@@ -836,6 +836,39 @@
...
+      "ERROR_SELECT_SOMETHING": "You have to select something",
+      "ERROR_FILL_REQUIRED": "Fill all required fields"
+    },
+    "DELETE_ATTRIBUTE_DEFINITION": {
+      "TITLE": "Confirm delete action",
+      "TEXT_LINE1": "Following attribute definitions will be removed.",
+      "TEXT_LINE2": "Do you want to proceed?",
+      "CANCEL": "Cancel",
+      "CONFIRM": "Delete",
+      "WARNING": "This action will will also delete all assigned values",
```
Comment (xkureck) [inline src/assets/i18n/en.json]: typo: "This action will will"

File: src/app/shared/components/dialogs/create-attribute-definition-dialog/create-attribute-definition-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    <mat-error
+      *ngIf="valueType === ''">{{'DIALOGS.CREATE_ATTRIBUTE_DEFINITION.ERROR_SELECT_SOMETHING' | translate}}</mat-error>
+  </mat-form-field>
+  <br>
+  <div class="d-flex flex-row mt-3">
+    <p>{{'DIALOGS.CREATE_ATTRIBUTE_DEFINITION.UNIQUE' | translate}}</p>
+    <p class="ml-4">OFF</p>
+    <mat-slide-toggle class="ml-2" color="accent" [(ngModel)]="attDef.unique">
+      ON
+    </mat-slide-toggle>
```
Comment (xkureck) [inline src/app/shared/components/dialogs/create-attribute-definition-dialog/create-attribute-definition-dialog.component.html]: When user pick "core" or "virt" definition attribute can't be unique. For now implementation allows it but after submiting perun return exception.
Also Entity "entityless" can't be Unique(Perun return's: "InternalErrorException Error 16dd986aa7e: entityless attributes cannot be marked unique")

File: src/app/shared/components/dialogs/delete-attribute-definition-dialog/delete-attribute-definition-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      <td mat-cell *matCellDef="let attribute"> {{attribute.displayName}} </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns" class="font-weight-bolder"></tr>
+    <tr mat-row *matRowDef="let group; columns: displayedColumns;"></tr>
+  </table>
+
+  <div style="background-color:#ff8282;" class="mt-3">
+    {{'DIALOGS.DELETE_ATTRIBUTE_DEFINITION.WARNING' | translate}}
+  </div>
```
Comment (xkureck) [inline src/app/shared/components/dialogs/delete-attribute-definition-dialog/delete-attribute-definition-dialog.component.html]: Vojta doesn't like when we mix html with css("style="background-color:#ff8282;"). It also shows warning in pretty weird way. Try something like "class=perun-alert warn-alert"".

**Comment by DanoFecko** [comment]
@xkureck to fix first issue you have to rebase back-end to newest version. I added 'deleteAttributes()' maybe a week ago.

**Comment by DanoFecko** [comment]
image

**Comment by DanoFecko** [comment]
@xkureck I fixed everything you found and some more :)

File: src/app/shared/components/dialogs/create-attribute-definition-dialog/create-attribute-definition-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+              private notificator: NotificatorService,
+              private translate: TranslateService,
+              private attributeService: AttributesService) {
+  }
+
+  attDef: AttributeDefinition;
+  entities: string[] = ['facility', 'resource', 'group', 'group_resource', 'host', 'member', 'member_group',
+    'member_resource', 'user', 'user_ext_source', 'user_facility', 'vo', 'entityless'];
+
+  definitionTypes: string[] = ['def', 'opt', 'virt', 'core'];
```
Comment (Vojtech-Sassmann) [inline src/app/shared/components/dialogs/create-attribute-definition-dialog/create-attribute-definition-dialog.component.ts]: Actually, the user should not be allowed to create a core attribute. In fact, if he does the call, the backend returns an OK status, but the attribute is not created. Core attributes reference entities' properties (like a name for a User) so they can be accessed the same way as any other information in the system.

**Comment by DanoFecko** [comment]
Travis has some problems with this build, I am not exactly sure what... I updated pom.xml and package.json and it didn't work.

**Pull request merged**